LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs

```diff
@@ -40,7 +41,10 @@ public void ConfigureServices(IServiceCollection services)
             _ = services.AddLogging(loggingBuilder =>
                         {
                             _ = loggingBuilder.ClearProviders();
-                            var logLevel = (LogLevel)int.Parse(NetworkServerConfiguration.LogLevel, CultureInfo.InvariantCulture);
+                            var logLevel = int.TryParse(NetworkServerConfiguration.LogLevel, out var logLevelNum)
+                                ? (LogLevel)logLevelNum
```

Comment:
C# allows integers conversions to enums to always succeed so use [`Enum.IsDefined`](https://docs.microsoft.com/en-us/dotnet/api/system.enum.isdefined) to ensure the parsed integer is indeed a defined value of `LogLevel`.

```suggestion
                                ? (LogLevel)logLevelNum is var level && Enum.IsDefined(typeof(LogLevel), level) ? level : throw new InvalidCastException()
```

Reply:
Done in b3c7f3921621aa11d8efa6036a95f8e43788fee6